segment_terraform.go

```diff
@@ -0,0 +1,24 @@
+package main
+
+type terraform struct {
+	props         *properties
+	env           environmentInfo
+	workspaceName string
+}
+
+func (tf *terraform) string() string {
+	return tf.workspaceName
+}
+
+func (tf *terraform) init(props *properties, env environmentInfo) {
+	tf.props = props
+	tf.env = env
+}
+
+func (tf *terraform) enabled() bool {
+	if !(tf.env.hasCommand("terraform") && tf.env.hasFolder(".terraform")) {
+		return false
+	}
+	tf.workspaceName = tf.env.runCommand("terraform", "workspace", "show")
```

Comment:
Can this fail when the previous conditions passes?

Reply:
Hm, no. The only requirement for the command to run is to have the terraform binary. It can run in all directories, even those who have nothing to do with Terraform. Even in directories that have not been subject to "terraform init" (i.e. they will not have a .terraform subdir) the command will output "default".

But it does not make sense to show this output in directories without the .terraform subdir, so that's why that check is added.

Reply:
A suggestion for how to include this test into segment_terraform_test.go (actually evaluate both cases together) would be of great help! :)